File: 0007-reverse-integer/0007-reverse-integer.py
class Solution:
    def reverse(self, x: int) -> int:
        def solve(n):
            nonlocal result
            
            if n > 0:
                result = result*10+(n%10)
                solve(n//10)

        
        neg = False
        
        if x < 0:
            x*=-1
            neg = True
            
        result = 0
        
        solve(x)
        
        if not -2**31<=result<=2**31-1:
            return 0
        
        return -1*result if neg else result

File: 0007-reverse-integer/0007-reverse-integer.cpp
class Solution {
public:
    long reverse(long x) {
        bool neg = false;
        
        if(x<0){
            neg = true;
            x*=-1;
        }
        
        long res = 0;
        
        while(x>0){
            res = res*10+(x%10);
            x/=10;
        }
        
        if((res > pow(2,31)-1) or (res <= pow(-2,31)))
            res = 0;
        return neg?-1*res:res;
        
    }
};